Userland/Utilities/unzip.cpp

```diff
@@ -124,7 +127,25 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     }
 
     auto success = zip_file->for_each_member([&](auto zip_member) {
-        return unpack_zip_member(zip_member, quiet) ? IterationDecision::Continue : IterationDecision::Break;
+        bool keep_file = false;
+
+        if (file_filters.size() != 0) {
+            for (auto& filter : file_filters) {
+                // Convert underscore wildcards (usual unzip convention) to question marks (as used by StringUtils)
+                String string_filter = AK::StringUtils::replace(filter, "_", "?");
+                if (AK::StringUtils::matches(zip_member.name, string_filter)) {
+                    keep_file = true;
+                    break;
+                }
+            }
+        } else {
+            keep_file = true;
+        }
+
+        if (keep_file && !unpack_zip_member(zip_member, quiet))
+            return IterationDecision::Break;
+
+        return IterationDecision::Continue;
```

Comment:
For clarity, I would nest these two conditions. I know it works this way, but it would be more understandable if you do this:
```suggestion
        if (keep_file) {
            if (!unpack_zip_member(zip_member, quiet))
                return IterationDecision::Break;
        }
        return IterationDecision::Continue;